server/tests/test_smoke.py

```diff
@@ -165,3 +184,96 @@ async def test_connection(client, caplog):
+    """
+
+    class MyLspType(lsp_types.LspModel):
+        snake_case_name: int
+        optional: Optional[int]
+
+    spec_compatible: Dict = {"snakeCaseName": 0}
+
+    v1 = MyLspType(snake_case_name=0)
+    v2 = MyLspType(snakeCaseName=0)
+    v3 = MyLspType.parse_obj(spec_compatible)
+
+    for v in [v1, v2, v3]:
+        assert v.dict() == spec_compatible
+
+
+@pytest.mark.timeout(5)
+@pytest.mark.asyncio
+async def test_symbol_provider(client: JsonRPC) -> None:
+    ret: int
+    result: object
+
+    await initialize_project(client, "project")
+
+    ret = await client.call("workspace/symbol", query="symbol")
+    result = await client.assert_one(ret)
+    assert isinstance(result, list)
+    symbol_info: List[lsp_types.SymbolInformation] = [lsp_types.SymbolInformation.parse_obj(symbol) for symbol in result]
+
+    project_dir: str = os.path.abspath(os.path.join(os.path.dirname(__file__), "project"))
+    uri_main: str = "file://%s" % os.path.join(project_dir, "main.cf")
+    testmodule_dir: str = os.path.join(project_dir, "libs", "testmodule")
+    uri_testmodule_model: str = "file://%s" % os.path.join(testmodule_dir, "model", "_init.cf")
+    uri_testmodule_plugins: str = "file://%s" % os.path.join(testmodule_dir, "plugins", "__init__.py")
+
+    assert symbol_info == [
+        lsp_types.SymbolInformation(
+            name="__config__::my_symbol_test_type",
+            kind=lsp_types.SymbolKind.Class,
+            location=lsp_types.Location(
+                uri=uri_main,
+                range=lsp_types.Range(
+                    start=lsp_types.Position(line=8, character=8), end=lsp_types.Position(line=8, character=27)
+                ),
+            ),
+        ),
+        lsp_types.SymbolInformation(
+            name="testmodule::SymbolTest",
+            kind=lsp_types.SymbolKind.Class,
+            location=lsp_types.Location(
+                uri=uri_testmodule_model,
+                range=lsp_types.Range(
+                    start=lsp_types.Position(line=0, character=7), end=lsp_types.Position(line=0, character=17)
+                ),
+            ),
+        ),
+        lsp_types.SymbolInformation(
+            name="testmodule::symbolTest",
+            kind=lsp_types.SymbolKind.Constructor,
+            location=lsp_types.Location(
+                uri=uri_testmodule_model,
+                range=lsp_types.Range(
+                    start=lsp_types.Position(line=4, character=15), end=lsp_types.Position(line=4, character=25)
+                ),
+            ),
+        ),
+        lsp_types.SymbolInformation(
+            name="testmodule::plugin_symbol_test",
+            kind=lsp_types.SymbolKind.Function,
+            location=lsp_types.Location(
+                uri=uri_testmodule_plugins,
+                range=lsp_types.Range(
+                    start=lsp_types.Position(line=4, character=0), end=lsp_types.Position(line=4, character=1)
```

Comment:
This range doesn't seem to be correct, is it?

Reply:
Alas, the location for a plugin is only a `Location`, not a `Range` (in the compiler).

Reply:
Ok